amadeus-core/src/par_sink/stddev.rs

```diff
@@ -0,0 +1,129 @@
+use derive_new::new;
+use educe::Educe;
+use serde::{Deserialize, Serialize};
+use std::marker::PhantomData;
+
+use super::{folder_par_sink, FolderSync, FolderSyncReducer, ParallelPipe, ParallelSink};
+
+#[derive(new)]
+#[must_use]
+pub struct StdDev<P> {
+	pipe: P,
+}
+
+impl_par_dist! {
+	impl<P: ParallelPipe<Item, Output = f64>, Item> ParallelSink<Item> for StdDev<P>
+		{
+		folder_par_sink!(
+			SDFolder<StepA>,
+			SDFolder<StepB>,
+			self,
+			SDFolder::new(),
+			SDFolder::new()
+		);
+	}
+}
+
+#[derive(Educe, Serialize, Deserialize, new)]
+#[educe(Clone)]
+#[serde(bound = "")]
+
+pub struct SDFolder<Step> {
+	marker: PhantomData<fn() -> Step>,
+}
+
+pub struct StepA;
+pub struct StepB;
+
+#[derive(Serialize, Deserialize, new)]
+pub struct SDState {
+	#[new(default)]
+	mean: f64,
+	#[new(default)]
+	correction: f64,
+	#[new(default)]
+	count: u64,
+	#[new(default)]
+	value: Vec<f64>,
```

Comment:
standard deviation can be calculated in an online manner (i.e. not allocating a `Vec`) and without pre-calculating the mean like [this](https://docs.rs/statrs/0.13.0/src/statrs/statistics/iter_statistics.rs.html#119-138) and [this](https://docs.rs/streaming-stats/0.2.3/src/stats/online.rs.html#65-74) (I haven't looked closely, I assume these are the same but if they aren't could you let me know the difference?).